Sources/Workspace/PinsStore.swift

```diff
@@ -61,27 +81,45 @@ public struct PinsStore {
     public var pins: AnySequence<Pin> {
         return AnySequence<Pin>(pinsMap.values)
     }
+    
+    /// Most recent error, if any.
+    /// FIXME: This needs to be an array of diagnostics, or perhaps a structured
+    /// log of what-all went on while trying to instantiate the PinStore.
+    public var error: Error?
+    
+    /// Returns true if and only iff there are any errors; when this property is
+    /// true, mutating PinStore operations are disallowed and will throw errors.
+    /// FIXME: We should have a protocol for "things that can hold errors", and
+    /// then the `hasErrors` property should be a convenience implementation so
+    /// it doesn't need to be implemented in every adopter of the protocol.
+    public var hasError: Bool { return error != nil }
```

Comment:
This seems reasonable to me, but the only thing I wonder about is whether the PinsStore type itself really needs to participate in this versus the client holding the PinsStore storing a Result<PinsStore, Error>...

That would force clients to acknowledge that they may have an error pin store, if they need to be structured in such a way that they return a live object which is in an error state.

It feels to me like a PinsStore isn't something I should need to have if it was in an error state when it loaded.